src/vecgf2.c

```diff
@@ -3237,16 +3232,13 @@ Int DistVecClosVec(
 	else
 	  {
 	    cnt=SumInt(cnt,one);
-	    vec = BLOCKS_GF2VEC(ovec);
-	    sum = BLOCKS_GF2VEC(osum);
+	    vec = CONST_BLOCKS_GF2VEC(ovec);
 	    SET_ELM_PLIST(d,di+1,cnt);
 	    CHANGED_BAG(d);
-	    chg=1;
 	  }
       }
-    AddGF2VecToGF2Vec(sum,BLOCKS_GF2VEC(ELM_PLIST(vp,i+1)),len);
+    AddGF2VecToGF2Vec(BLOCKS_GF2VEC(osum),CONST_BLOCKS_GF2VEC(ELM_PLIST(vp,i+1)),len);
```

Comment:
Just checking, what's happening here?

Reply:
So, there is `Obj osum` (I guess the `o` in `osum` indicates that it is the "object"), and `UInt * sum` was set via `sum = BLOCKS_GF2VEC(osum);` -- and then there was logic to reset `sum` if `chg` is non-zero. And `chg` was tracked through recursion, presumably to track whether there *might* have been a GC. I decided that all this complexity is not worth the potential minor saving (I am not even sure it exists, because the code to track the `chg` state also takes up resources).

If one is concerned about optimizing this code, the first step should instead be to replace `SumInt` by  a suitably application of `ARE_INTOBJS` and `SUM_INTOBJS`.